libfoundation/src/foundation-string.cpp

```diff
@@ -6963,6 +6963,60 @@ bool MCStringGetNumericValue(MCStringRef self, double &r_value)
         return false;
 }
 
+MC_DLLEXPORT
+bool MCStringNormalizeLineEndings(MCStringRef p_input, 
+                                  MCStringLineEndingStyle p_to_style, 
+                                  MCStringRef& r_output, 
+                                  MCStringLineEndingStyle* r_original_style)
+{
+    uindex_t t_firstLF = 0;
+    uindex_t t_firstCR = 0;
+    MCStringLineEndingStyle t_original_style = kMCStringLineEndingStyleLF;
+    MCStringRef t_mutable_input;
+    
+    if (MCStringFirstIndexOfChar(p_input, '\r', 0, kMCStringOptionCompareExact, t_firstCR))
+    {
+        if (MCStringFirstIndexOfChar(p_input, '\n', 0, kMCStringOptionCompareExact, t_firstLF))
+        {
+            if ((t_firstCR + 1 == t_firstLF) || (t_firstLF + 1 == t_firstCR))
+                t_original_style = kMCStringLineEndingStyleCRLF;
+            else if (t_firstCR < t_firstLF)
+                t_original_style = kMCStringLineEndingStyleCR;
+        }
+        else
+            t_original_style = kMCStringLineEndingStyleCR;
+    }
+    
+    // normalize input to LF line endings first
+    /* UNCHECKED */ MCStringMutableCopy(p_input, t_mutable_input);
```

Comment:
Move the declaration of t_mutable_input to here and use an auto-class:
```
MCAutoStringRef t_mutable_input;
```
You can then easily make all the calls checked:
```
if (!MCStringMutableCopy(p_input, &t_mutable_input))
{
  return false;
}
```